portal-api/src/v1/controllers/utilisation-report-service/utilisation-report-upload.controller.js

```diff
@@ -108,34 +109,37 @@ const uploadReportAndSendNotification = async (req, res) => {
       return res.status(400).send();
     }
 
-    // If a report has already been uploaded, we should not overwrite it
-    const uploadedReportsInReportPeriod = await api.getUtilisationReports(parsedUser?.bank?.id, {
+    const bankId = parsedUser?.bank?.id;
+
+    const existingReports = await api.getUtilisationReports(bankId, {
       reportPeriod: parsedReportPeriod,
-      excludeNotUploaded: true,
     });
-    if (uploadedReportsInReportPeriod.length > 0) {
+
+    if (existingReports.length !== 1) {
+      return res.status(500).send(`Expected 1 report but found ${existingReports.length} with bank ID ${bankId} and report period '${parsedReportPeriod}'`);
+    }
+
+    const existingReport = existingReports[0];
+
+    // If a report has already been uploaded, we should not overwrite it
+    if (existingReport.azureFileInfo) {
       return res.status(409).send('Report for the supplied report period has already been uploaded');
     }
 
-    const fileInfo = await saveFileToAzure(file, parsedUser.bank.id);
+    const fileInfo = await saveFileToAzure(file, bankId);
```

Comment:
This section here -  are we trying to allow a user to upload a report if it's 'Marked as complete'?
If it's marked as complete it means they shouldnt be able to upload a report for that period. I'd prefer this to actually check that there are no reports that aren't in the 'NOT_RECEIVED' status

Reply:
Alright - I will update to check for status instead of whether or not a report has been uploaded